### PR TITLE
frontend: auth + other misc fixes

### LINKED ISSUE #72: [question-service] inconsistent question-duplicate handling
gonna put this somewhere before i forget

- add questions disallows duplicate titles
- edit question doesnt care

### PR DESCRIPTION
**Pull request by seelengxd**
### auth related
- [x] [backend - user-service] send user over after login or frontend will crash on accessing question
- [x] render a 403 page ripped off the 404 page
- [x] conditionally hide create/edit/delete buttons if you arent an admin
- [x] /currentUser should attempt /token if it fails

### random stuff
- [x] remove duplicate description in viewquestion
- [x] redirect to 404 if question not found view/edit
- [x] give minWidth to ViewQuestion description or it looks a bit sus if the description is small
- [x] fix edit question duplicate check

## pr will not handle
- go chase every api call to attempt /token when unauthorised

### CODE REVIEW

File: frontend/src/pages/questions/UpdateQuestion.tsx [hunk 1]
```diff
@@ -1,11 +1,14 @@
 import React, { useEffect, useState } from 'react';
 import QuestionsAPI from '../../api/questions/questions';
-import { useParams } from 'react-router-dom';
+import { useNavigate, useParams } from 'react-router-dom';
 import QuestionForm from '../../components/content/QuestionForm';
 import { LINK_PREFIX, type QuestionPostData, type QuestionData } from '../../types/questions/questions';
 
 export const UpdateQuestion: React.FC = () => {
+  console.log('huh?');
```
Comment (sltsheryl) [inline frontend/src/pages/questions/UpdateQuestion.tsx]: can remove

**Review by sltsheryl — APPROVED** [review]
Screenshot 2023-09-30 at 3 34 46 PM

LGTM. just a question, if a user accesses a protected route on the login page, should we show a 404 or no change is better

File: frontend/src/pages/questions/UpdateQuestion.tsx (hunk 1 above)
Comment (seelengxd) [reply frontend/src/pages/questions/UpdateQuestion.tsx]: oops 🤡 thanks for the catch

**Comment by seelengxd** [comment]
hmm i think we should still show login page... how about we also redirect to /login so the url looks less sus? :P 

image

**Pull request merged**